### PR TITLE
rust-wasm-cross: fix darwin and aarch64-linux builds

### PR DESCRIPTION
**Pull request by sandydoo**
- Fix the linker issue on macOS. Link to upstream issue.
- Fix the build on `aarch64-linux`, where `wasm-pack` complains that it can't load a build of `wasm-opt`.

### CODE REVIEW

File: examples/rust-wasm-cross/.test.sh [hunk 1]
```diff
@@ -4,7 +4,8 @@ cargo --version
 rustc --version
 
 if [[ "$(uname)" == "Darwin" ]]; then
-  echo "$RUSTFLAGS" | grep -- "-L framework=$DEVENV_PROFILE/Library/Frameworks"
+  # See env.RUSTFLAGS in devenv.nix
+  # echo "$RUSTFLAGS" | grep -- "-L framework=$DEVENV_PROFILE/Library/Frameworks"
```
Comment (domenkozar) [inline examples/rust-wasm-cross/.test.sh:8]: we should still assert this when not doing wasm

Comment (sandydoo) [reply examples/rust-wasm-cross/.test.sh:8]: This example is wasm-only though

Comment (domenkozar) [reply examples/rust-wasm-cross/.test.sh:8]: Ah! Would it make sense to just remove this?

Comment (sandydoo) [reply examples/rust-wasm-cross/.test.sh:8]: I guess this got copy-pasted over from the regular rust test. 🤷

**Comment by sandydoo** [comment]
Linking the upstream issue, so we can update this when that's resolved: https://github.com/rust-lang/rust/issues/122333

**Pull request merged**